meson.build

```diff
@@ -0,0 +1,97 @@
+project(
+ 'jsoncpp',
+ 'cpp',
+ version : '1.8.0',
+ default_options : [
+  'buildtype=release',
+  'cpp_std=c++11',
+  'warning_level=1'],
+ license : 'Public Domain',
+ meson_version : '>= 0.41.1')
+
+jsoncpp_ver_arr = meson.project_version().split('.')
+jsoncpp_major_version = jsoncpp_ver_arr[0]
+jsoncpp_minor_version = jsoncpp_ver_arr[1]
+jsoncpp_patch_version = jsoncpp_ver_arr[2]
+
+jsoncpp_cdata = configuration_data()
+jsoncpp_cdata.set('JSONCPP_VERSION', meson.project_version())
+jsoncpp_cdata.set('JSONCPP_VERSION_MAJOR', jsoncpp_major_version)
+jsoncpp_cdata.set('JSONCPP_VERSION_MINOR', jsoncpp_minor_version)
+jsoncpp_cdata.set('JSONCPP_VERSION_PATCH', jsoncpp_patch_version)
+
+jsoncpp_version_h = configure_file(
+ input : 'src/lib_json/version.h.in',
+ output : 'version.h',
+ configuration : jsoncpp_cdata,
+ install : true,
+ install_dir : join_paths(get_option('prefix'), get_option('includedir'), 'json')
+)
+
+jsoncpp_headers = [
+ 'include/json/allocator.h',
+ 'include/json/assertions.h',
+ 'include/json/autolink.h',
+ 'include/json/config.h',
+ 'include/json/features.h',
+ 'include/json/forwards.h',
+ 'include/json/json.h',
+ 'include/json/reader.h',
+ 'include/json/value.h',
+ 'include/json/writer.h']
+jsoncpp_include_directories = include_directories('include')
+
+install_headers(
+ jsoncpp_headers,
+ subdir : 'json')
+
+jsoncpp_lib = library(
+ 'jsoncpp',
+ [jsoncpp_version_h,
+  jsoncpp_headers,
+  'src/lib_json/json_tool.h',
+  'src/lib_json/json_reader.cpp',
+  'src/lib_json/json_value.cpp',
+  'src/lib_json/json_writer.cpp'],
+ soversion : 18,
+ install : true,
+ include_directories : jsoncpp_include_directories)
+
+pkg = import('pkgconfig')
+pkg.generate(
+ libraries : jsoncpp_lib,
+ version : meson.project_version(),
+ name : 'jsoncpp',
+ filebase : 'jsoncpp',
+ description : 'A C++ library for interacting with JSON')
+
+# tests
+python3 = import('python3').find_python()
+
+jsoncpp_test = executable(
+ 'jsoncpp_test',
+ ['src/test_lib_json/jsontest.cpp',
+  'src/test_lib_json/jsontest.h',
+  'src/test_lib_json/main.cpp'],
+ include_directories : jsoncpp_include_directories,
+ link_with : jsoncpp_lib,
+ install : false)
+test(
+ 'unittest_jsoncpp_test',
+ jsoncpp_test)
+
+jsontestrunner = executable(
+ 'jsontestrunner',
+ 'src/jsontestrunner/main.cpp',
+ include_directories : jsoncpp_include_directories,
+ link_with : jsoncpp_lib,
+ install : false)
+test(
+ 'unittest_jsontestrunner',
+ python3,
+ args : [
+  '-B',
+  join_paths(meson.current_source_dir(), 'test/runjsontests.py'),
+  jsontestrunner,
+  join_paths(meson.current_source_dir(), 'test/data')]
```

Comment:
```
$ meson --default-library shared . build
<snip>
$ ninja -C build test
ninja: Entering directory `build'
[10/11] Running all tests.
1/2 unittest_jsoncpp_test                   OK       0.01 s
2/2 unittest_jsontestrunner                 OK       1.05 s

OK:         2
FAIL:       0
SKIP:       0
TIMEOUT:    0

Full log written to /home/soap/gitz/jsoncpp/build/meson-logs/testlog.txt
$ git clean -fdx .
Removing build/
Removing test/data/test_array_01.actual
Removing test/data/test_array_01.actual-rewrite
Removing test/data/test_array_01.process-output
Removing test/data/test_array_01.rewrite
Removing test/data/test_array_02.actual
Removing test/data/test_array_02.actual-rewrite
Removing test/data/test_array_02.process-output
Removing test/data/test_array_02.rewrite
Removing test/data/test_array_03.actual
Removing test/data/test_array_03.actual-rewrite
Removing test/data/test_array_03.process-output
Removing test/data/test_array_03.rewrite
Removing test/data/test_array_04.actual
Removing test/data/test_array_04.actual-rewrite
Removing test/data/test_array_04.process-output
Removing test/data/test_array_04.rewrite
Removing test/data/test_array_05.actual
Removing test/data/test_array_05.actual-rewrite
Removing test/data/test_array_05.process-output
Removing test/data/test_array_05.rewrite
Removing test/data/test_array_06.actual
Removing test/data/test_array_06.actual-rewrite
Removing test/data/test_array_06.process-output
Removing test/data/test_array_06.rewrite
Removing test/data/test_array_07.actual
Removing test/data/test_array_07.actual-rewrite
Removing test/data/test_array_07.process-output
Removing test/data/test_array_07.rewrite
Removing test/data/test_basic_01.actual
Removing test/data/test_basic_01.actual-rewrite
Removing test/data/test_basic_01.process-output
Removing test/data/test_basic_01.rewrite
Removing test/data/test_basic_02.actual
Removing test/data/test_basic_02.actual-rewrite
Removing test/data/test_basic_02.process-output
Removing test/data/test_basic_02.rewrite
Removing test/data/test_basic_03.actual
Removing test/data/test_basic_03.actual-rewrite
Removing test/data/test_basic_03.process-output
Removing test/data/test_basic_03.rewrite
Removing test/data/test_basic_04.actual
Removing test/data/test_basic_04.actual-rewrite
Removing test/data/test_basic_04.process-output
Removing test/data/test_basic_04.rewrite
Removing test/data/test_basic_05.actual
Removing test/data/test_basic_05.actual-rewrite
Removing test/data/test_basic_05.process-output
Removing test/data/test_basic_05.rewrite
Removing test/data/test_basic_06.actual
Removing test/data/test_basic_06.actual-rewrite
Removing test/data/test_basic_06.process-output
Removing test/data/test_basic_06.rewrite
Removing test/data/test_basic_07.actual
Removing test/data/test_basic_07.actual-rewrite
Removing test/data/test_basic_07.process-output
Removing test/data/test_basic_07.rewrite
Removing test/data/test_basic_08.actual
Removing test/data/test_basic_08.actual-rewrite
Removing test/data/test_basic_08.process-output
Removing test/data/test_basic_08.rewrite
Removing test/data/test_basic_09.actual
Removing test/data/test_basic_09.actual-rewrite
Removing test/data/test_basic_09.process-output
Removing test/data/test_basic_09.rewrite
Removing test/data/test_comment_00.actual
Removing test/data/test_comment_00.actual-rewrite
Removing test/data/test_comment_00.process-output
Removing test/data/test_comment_00.rewrite
Removing test/data/test_comment_01.actual
Removing test/data/test_comment_01.actual-rewrite
Removing test/data/test_comment_01.process-output
Removing test/data/test_comment_01.rewrite
Removing test/data/test_comment_02.actual
Removing test/data/test_comment_02.actual-rewrite
Removing test/data/test_comment_02.process-output
Removing test/data/test_comment_02.rewrite
Removing test/data/test_complex_01.actual
Removing test/data/test_complex_01.actual-rewrite
Removing test/data/test_complex_01.process-output
Removing test/data/test_complex_01.rewrite
Removing test/data/test_integer_01.actual
Removing test/data/test_integer_01.actual-rewrite
Removing test/data/test_integer_01.process-output
Removing test/data/test_integer_01.rewrite
Removing test/data/test_integer_02.actual
Removing test/data/test_integer_02.actual-rewrite
Removing test/data/test_integer_02.process-output
Removing test/data/test_integer_02.rewrite
Removing test/data/test_integer_03.actual
Removing test/data/test_integer_03.actual-rewrite
Removing test/data/test_integer_03.process-output
Removing test/data/test_integer_03.rewrite
Removing test/data/test_integer_04.actual
Removing test/data/test_integer_04.actual-rewrite
Removing test/data/test_integer_04.process-output
Removing test/data/test_integer_04.rewrite
Removing test/data/test_integer_05.actual
Removing test/data/test_integer_05.actual-rewrite
Removing test/data/test_integer_05.process-output
Removing test/data/test_integer_05.rewrite
Removing test/data/test_integer_06_64bits.actual
Removing test/data/test_integer_06_64bits.actual-rewrite
Removing test/data/test_integer_06_64bits.process-output
Removing test/data/test_integer_06_64bits.rewrite
Removing test/data/test_integer_07_64bits.actual
Removing test/data/test_integer_07_64bits.actual-rewrite
Removing test/data/test_integer_07_64bits.process-output
Removing test/data/test_integer_07_64bits.rewrite
Removing test/data/test_integer_08_64bits.actual
Removing test/data/test_integer_08_64bits.actual-rewrite
Removing test/data/test_integer_08_64bits.process-output
Removing test/data/test_integer_08_64bits.rewrite
Removing test/data/test_large_01.actual
Removing test/data/test_large_01.actual-rewrite
Removing test/data/test_large_01.process-output
Removing test/data/test_large_01.rewrite
Removing test/data/test_object_01.actual
Removing test/data/test_object_01.actual-rewrite
Removing test/data/test_object_01.process-output
Removing test/data/test_object_01.rewrite
Removing test/data/test_object_02.actual
Removing test/data/test_object_02.actual-rewrite
Removing test/data/test_object_02.process-output
Removing test/data/test_object_02.rewrite
Removing test/data/test_object_03.actual
Removing test/data/test_object_03.actual-rewrite
Removing test/data/test_object_03.process-output
Removing test/data/test_object_03.rewrite
Removing test/data/test_object_04.actual
Removing test/data/test_object_04.actual-rewrite
Removing test/data/test_object_04.process-output
Removing test/data/test_object_04.rewrite
Removing test/data/test_preserve_comment_01.actual
Removing test/data/test_preserve_comment_01.actual-rewrite
Removing test/data/test_preserve_comment_01.process-output
Removing test/data/test_preserve_comment_01.rewrite
Removing test/data/test_real_01.actual
Removing test/data/test_real_01.actual-rewrite
Removing test/data/test_real_01.process-output
Removing test/data/test_real_01.rewrite
Removing test/data/test_real_02.actual
Removing test/data/test_real_02.actual-rewrite
Removing test/data/test_real_02.process-output
Removing test/data/test_real_02.rewrite
Removing test/data/test_real_03.actual
Removing test/data/test_real_03.actual-rewrite
Removing test/data/test_real_03.process-output
Removing test/data/test_real_03.rewrite
Removing test/data/test_real_04.actual
Removing test/data/test_real_04.actual-rewrite
Removing test/data/test_real_04.process-output
Removing test/data/test_real_04.rewrite
Removing test/data/test_real_05.actual
Removing test/data/test_real_05.actual-rewrite
Removing test/data/test_real_05.process-output
Removing test/data/test_real_05.rewrite
Removing test/data/test_real_06.actual
Removing test/data/test_real_06.actual-rewrite
Removing test/data/test_real_06.process-output
Removing test/data/test_real_06.rewrite
Removing test/data/test_real_07.actual
Removing test/data/test_real_07.actual-rewrite
Removing test/data/test_real_07.process-output
Removing test/data/test_real_07.rewrite
Removing test/data/test_real_08.actual
Removing test/data/test_real_08.actual-rewrite
Removing test/data/test_real_08.process-output
Removing test/data/test_real_08.rewrite
Removing test/data/test_real_09.actual
Removing test/data/test_real_09.actual-rewrite
Removing test/data/test_real_09.process-output
Removing test/data/test_real_09.rewrite
Removing test/data/test_real_10.actual
Removing test/data/test_real_10.actual-rewrite
Removing test/data/test_real_10.process-output
Removing test/data/test_real_10.rewrite
Removing test/data/test_real_11.actual
Removing test/data/test_real_11.actual-rewrite
Removing test/data/test_real_11.process-output
Removing test/data/test_real_11.rewrite
Removing test/data/test_real_12.actual
Removing test/data/test_real_12.actual-rewrite
Removing test/data/test_real_12.process-output
Removing test/data/test_real_12.rewrite
Removing test/data/test_string_01.actual
Removing test/data/test_string_01.actual-rewrite
Removing test/data/test_string_01.process-output
Removing test/data/test_string_01.rewrite
Removing test/data/test_string_02.actual
Removing test/data/test_string_02.actual-rewrite
Removing test/data/test_string_02.process-output
Removing test/data/test_string_02.rewrite
Removing test/data/test_string_03.actual
Removing test/data/test_string_03.actual-rewrite
Removing test/data/test_string_03.process-output
Removing test/data/test_string_03.rewrite
Removing test/data/test_string_04.actual
Removing test/data/test_string_04.actual-rewrite
Removing test/data/test_string_04.process-output
Removing test/data/test_string_04.rewrite
Removing test/data/test_string_05.actual
Removing test/data/test_string_05.actual-rewrite
Removing test/data/test_string_05.process-output
Removing test/data/test_string_05.rewrite
Removing test/data/test_string_unicode_01.actual
Removing test/data/test_string_unicode_01.actual-rewrite
Removing test/data/test_string_unicode_01.process-output
Removing test/data/test_string_unicode_01.rewrite
Removing test/data/test_string_unicode_02.actual
Removing test/data/test_string_unicode_02.actual-rewrite
Removing test/data/test_string_unicode_02.process-output
Removing test/data/test_string_unicode_02.rewrite
Removing test/data/test_string_unicode_03.actual
Removing test/data/test_string_unicode_03.actual-rewrite
Removing test/data/test_string_unicode_03.process-output
Removing test/data/test_string_unicode_03.rewrite
Removing test/data/test_string_unicode_04.actual
Removing test/data/test_string_unicode_04.actual-rewrite
Removing test/data/test_string_unicode_04.process-output
Removing test/data/test_string_unicode_04.rewrite
Removing test/data/test_string_unicode_05.actual
Removing test/data/test_string_unicode_05.actual-rewrite
Removing test/data/test_string_unicode_05.process-output
Removing test/data/test_string_unicode_05.rewrite
```
All those temporary files in `test/` should not be created there but in `build/`, as the idea of meson is not to use `make clean` and just `rm -rf build/`. You might want to consider chopping up that python script and turning it into separate parallel tests.